### PR TITLE
Major improvements on visual tools and new kernels added

### PR DESCRIPTION
**Pull request by aster-stoustrup**
Added new visual tools, cleaned up most of the old ones. Importantly, normalisation can now be turned off for all graphs and this is done automatically.

Various fixes and other smaller features. See changelog for more details.

### CODE REVIEW

File: veropt/optimiser/model.py [hunk 1]
```diff
@@ -1003,42 +761,52 @@ def _train_backwards(self) -> None:
...
+                iteration += 1
+                if iteration > self.settings.max_iter:
+                    warnings.warn("Stopped training due to maximum iterations reached.")
+                    break
 
-        if self.training_settings.verbose:
+        if self.settings.verbose:
             print("\n")
 
     def _initiate_optimiser(self) -> None:
```
Comment (mrozowskamartaa) [inline veropt/optimiser/model.py]: change this name to initialise too for consistency

File: veropt/interfaces/utility.py [hunk 2]
```diff
@@ -60,8 +81,16 @@ def save_to_json(
             config_file: str
     ) -> None:
 
-        with open(config_file, "w") as f:
-            f.write(self.model_dump_json())
+        with open(config_file, "w") as json_file:
+
+            json_dict = self.model_dump(mode='json')
+            json_dict = _replace_nans_in_json_dict(json_dict)  # Necessary because of pydantic bug >:)
```
Comment (mrozowskamartaa) [inline veropt/interfaces/utility.py]: Hmmm... since there's already a method for validating dataclass dicts in the optimiser, maybe we could just get rid of pydantic and write the functionality we want (which is so far a small subset of what pydantic offers) ourselves?

File: veropt/optimiser/model.py (hunk 1 above)
Comment (aster-stoustrup) [reply veropt/optimiser/model.py]: Should be fixed now :))
